### PR TITLE
Fix: `TopicMessageQuery` refactor issues

### PR DESCRIPTION
**Pull request by danielakhterov**

### CODE REVIEW

**Review by cacampbell — APPROVED** [review]

**Comment by danielakhterov** [comment]
Merge as this is an important fix that needs to be released ASAP. However, I'd still like any reviewers to thoroughly review my changes in case I've made more silly mistakes.
Note: CI is failing on node, but passing on chrome. I've ran locally and it seems to be passing as well. I'm guessing simply increasing the timeout for the particular test case that is failing should resolve the issue, but that's for another PR.

File: src/topic/TopicMessageQuery.js [hunk 1]
```diff
@@ -332,6 +341,16 @@ export default class TopicMessageQuery {
                 (data) => {
                     const message = proto.ConsensusTopicResponse.decode(data);
 
+                    if (this._limit != null && this._limit.gt(0)) {
+                        this._limit = this._limit.sub(1);
+                    }
+
+                    this._startTime = Timestamp._fromProtobuf(
```
Comment (steven-sheehy) [inline src/topic/TopicMessageQuery.js:348]: Need to add one nanosecond

File: src/topic/TopicMessageQuery.js [hunk 2]
```diff
@@ -388,9 +403,10 @@ export default class TopicMessageQuery {
 
                         setTimeout(() => {
```
Comment (steven-sheehy) [inline src/topic/TopicMessageQuery.js:404]: There's still no log that indicates a retry is in progress

File: src/topic/TopicMessageQuery.js (hunk 1 above)
Comment (danielakhterov) [reply src/topic/TopicMessageQuery.js:348]: Damn, good catch. Fixed this in https://github.com/hashgraph/hedera-sdk-js/pull/411

File: src/topic/TopicMessageQuery.js (hunk 2 above)
Comment (danielakhterov) [reply src/topic/TopicMessageQuery.js:404]: I believe I also forgot to add logging in Go; logging just isn't second nature to me yet. Fixed here though https://github.com/hashgraph/hedera-sdk-js/pull/411

**Pull request merged**